src/API/Google/AdsAsset.php

```diff
@@ -93,16 +117,109 @@ protected function get_asset_type_by_field_type( string $field_type ): int {
 
 	}
 
+	/**
+	 * Returns the image data.
+	 *
+	 * @param string $url The image url.
+	 *
+	 * @return array The image data.
+	 * @throws Exception If the image url is not a valid url or the image size is too large.
+	 */
+	protected function get_image_data( string $url ): array {
+		$image_data = $this->wp->wp_remote_get( $url );
+
+		if ( is_wp_error( $image_data ) || empty( $image_data['body'] ) ) {
+			throw new Exception( 'Incorrect image asset url.' );
```

Comment:
The first time I tested I was getting Curl remote error timeouts here. Which was not really an  `Incorrect image asset url`. After debugging I saw the CURL error but I was thinking it was a problem in the URLs. 

I would change this Exception to other thing (maybe getting the error?) or at least including the url could help more.